src/refdb_backend.c

```diff
@@ -0,0 +1,419 @@
+        be->del = PyObject_GetAttrString((PyObject *)self, "del");
+        be->backend.del = pygit2_refdb_backend_del;
+        Py_INCREF(be->del);
+    }
+
+    if (PyObject_HasAttrString((PyObject *)self, "compress")) {
+        be->compress = PyObject_GetAttrString((PyObject *)self, "compress");
+        be->backend.compress = pygit2_refdb_backend_compress;
+        Py_INCREF(be->compress);
+    }
+    */
+
+    /* TODO: reflog
+    if (PyObject_HasAttrString((PyObject *)self, "has_log")) {
+        be->has_log = PyObject_GetAttrString((PyObject *)self, "has_log");
+        be->backend.has_log = pygit2_refdb_backend_has_log;
+        Py_INCREF(be->has_log);
+    }
+
+    if (PyObject_HasAttrString((PyObject *)self, "ensure_log")) {
+        be->ensure_log = PyObject_GetAttrString((PyObject *)self, "ensure_log");
+        be->backend.ensure_log = pygit2_refdb_backend_ensure_log;
+        Py_INCREF(be->ensure_log);
+    }
+
+    if (PyObject_HasAttrString((PyObject *)self, "reflog_read")) {
+        be->reflog_read = PyObject_GetAttrString((PyObject *)self, "reflog_read");
+        be->backend.reflog_read = pygit2_refdb_backend_reflog_read;
+        Py_INCREF(be->reflog_read);
+    }
+
+    if (PyObject_HasAttrString((PyObject *)self, "reflog_write")) {
+        be->reflog_write = PyObject_GetAttrString((PyObject *)self, "reflog_write");
+        be->backend.reflog_write = pygit2_refdb_backend_reflog_write;
+        Py_INCREF(be->reflog_write);
+    }
+
+    if (PyObject_HasAttrString((PyObject *)self, "reflog_rename")) {
+        be->reflog_rename = PyObject_GetAttrString((PyObject *)self, "reflog_rename");
+        be->backend.reflog_write = pygit2_refdb_backend_reflog_rename;
+        Py_INCREF(be->reflog_rename);
+    }
+
+    if (PyObject_HasAttrString((PyObject *)self, "reflog_delete")) {
+        be->reflog_delete = PyObject_GetAttrString((PyObject *)self, "reflog_delete");
+        be->backend.reflog_delete = pygit2_refdb_backend_reflog_delete;
+        Py_INCREF(be->reflog_delete);
+    }
+    */
+
+    /* TODO
+    if (PyObject_HasAttrString((PyObject *)self, "lock")) {
+        be->lock = PyObject_GetAttrString((PyObject *)self, "lock");
+        be->backend.lock = pygit2_refdb_backend_lock;
+        Py_INCREF(be->lock);
+    }
+
+    if (PyObject_HasAttrString((PyObject *)self, "unlock")) {
+        be->unlock = PyObject_GetAttrString((PyObject *)self, "unlock");
+        be->backend.unlock = pygit2_refdb_backend_unlock;
+        Py_INCREF(be->unlock);
+    }
+    */
+
+    Py_INCREF((PyObject *)self);
+    be->backend.free = pygit2_refdb_backend_free;
+
+    self->refdb_backend = (git_refdb_backend *)be;
+    return 0;
+}
+
+void
+RefdbBackend_dealloc(RefdbBackend *self)
+{
+    if (self->refdb_backend && self->refdb_backend->free == pygit2_refdb_backend_free) {
+        struct pygit2_refdb_backend *be = (struct pygit2_refdb_backend *)self->refdb_backend;
+        free(be);
```

Comment:
Here we need to clear as well the references to the methods, like it's done in the object db backend code: `Py_CLEAR(be->exists)` etc.